test/TestCase.h

```diff
@@ -68,21 +105,21 @@ class TestCase
 
 	static bool isTestFilename(boost::filesystem::path const& _filename);
 
-	/// Validates the settings, i.e. moves them from m_settings to m_validatedSettings.
-	/// Throws a runtime exception if any setting is left at this class (i.e. unknown setting).
 	/// Returns true, if the test case is supported in the current environment and false
 	/// otherwise which causes this test to be skipped.
 	/// This might check e.g. for restrictions on the EVM version.
-	virtual bool validateSettings(langutil::EVMVersion /*_evmVersion*/);
+	/// The function throws an exception if there are unread settings.
+	bool shouldRun();
 
 protected:
-	std::pair<std::map<std::string, std::string>, std::size_t> parseSourcesAndSettingsWithLineNumbers(std::istream& _file);
-	std::map<std::string, std::string> parseSourcesAndSettings(std::istream& _file);
-	std::pair<std::string, std::size_t> parseSourceAndSettingsWithLineNumbers(std::istream& _file);
-	std::string parseSourceAndSettings(std::istream& _file);
-	static void expect(std::string::iterator& _it, std::string::iterator _end, std::string::value_type _c);
+	// Used by ASTJSONTest, the only TestCase class with a custom parser of the test files.
+	TestCase() = default;
 
-	static std::string parseSimpleExpectations(std::istream& _file);
+	TestCase(std::string const& _filename): m_reader(_filename) {}
+
+	TestCaseReader m_reader;
```

Comment:
Please put all members at the very bottom.

Reply:
Sure, 6c9736b.